lib/node_modules/@stdlib/blas/base/sspr/benchmark/c/benchmark.length.c

```diff
@@ -0,0 +1,180 @@
+
+#define NAME "sspr"
+#define ITERATIONS 10000000
+#define REPEATS 3
+#define MIN 1
+#define MAX 6
+
+/**
+* Prints the TAP version.
+*/
+static void print_version( void ) {
+	printf( "TAP version 13\n" );
+}
+
+/**
+* Prints the TAP summary.
+*
+* @param total     total number of tests
+* @param passing   total number of passing tests
+*/
+static void print_summary( int total, int passing ) {
+	printf( "#\n" );
+	printf( "1..%d\n", total ); // TAP plan
+	printf( "# total %d\n", total );
+	printf( "# pass  %d\n", passing );
+	printf( "#\n" );
+	printf( "# ok\n" );
+}
+
+/**
+* Prints benchmarks results.
+*
+* @param iterations   number of iterations
+* @param elapsed      elapsed time in seconds
+*/
+static void print_results( int iterations, double elapsed ) {
+	double rate = (double)iterations / elapsed;
+	printf( "  ---\n" );
+	printf( "  iterations: %d\n", iterations );
+	printf( "  elapsed: %0.9f\n", elapsed );
+	printf( "  rate: %0.9f\n", rate );
+	printf( "  ...\n" );
+}
+
+/**
+* Returns a clock time.
+*
+* @return clock time
+*/
+static double tic( void ) {
+	struct timeval now;
+	gettimeofday( &now, NULL );
+	return (double)now.tv_sec + (double)now.tv_usec/1.0e6;
+}
+
+/**
+* Runs a benchmark.
+*
+* @param iterations   number of iterations
+* @param len          array length
+* @return elapsed time in seconds
+*/
+static double benchmark1( int iterations, int len ) {
+	double elapsed;
+	float AP[ len*(len+1)/2 ];
+	float x[ len ];
+	double t;
+	int i;
+
+	stdlib_strided_sfill( len, 0.5f, x, 1 );
+	stdlib_strided_sfill( len*(len+1)/2, 1.0f, AP, 1 );
+	t = tic();
+	for ( i = 0; i < iterations; i++ ) {
+		c_sspr( CblasRowMajor, CblasUpper, len, 1.0, x, 1, AP );
+		if ( AP[ 0 ] != AP[ 0 ] ) {
+			printf( "should not return NaN\n" );
+			break;
+		}
+	}
+	elapsed = tic() - t;
+	if ( AP[ 0 ] != AP[ 0 ] ) {
+		printf( "should not return NaN\n" );
+	}
+	return elapsed;
+}
+
+/**
+* Runs a benchmark.
+*
+* @param iterations   number of iterations
+* @param len          array length
+* @return elapsed time in seconds
+*/
+static double benchmark2( int iterations, int len ) {
+	double elapsed;
+	float AP[ len*(len+1)/2 ];
+	float x[ len ];
+	double t;
+	int i;
+
+	stdlib_strided_sfill( len, 0.5f, x, 1 );
+	stdlib_strided_sfill( len*(len+1)/2, 1.0f, AP, 1 );
+	t = tic();
+	for ( i = 0; i < iterations; i++ ) {
+		c_sspr_ndarray( CblasRowMajor, CblasUpper, len, 1.0, x, 1, 0, AP, 1, 0 );
+		if ( AP[ 0 ] != AP[ 0 ] ) {
+			printf( "should not return NaN\n" );
+			break;
+		}
+	}
+	elapsed = tic() - t;
+	if ( AP[ 0 ] != AP[ 0 ] ) {
+		printf( "should not return NaN\n" );
+	}
+	return elapsed;
+}
+
+/**
+* Main execution sequence.
+*/
+int main( void ) {
+	double elapsed;
+	int count;
+	int iter;
+	int len;
+	int i;
+	int j;
+
+	// Use the current time to seed the random number generator:
+	srand( time( NULL ) );
+
+	print_version();
+	count = 0;
+	for ( i = MIN; i <= MAX; i++ ) {
+		len = stdlib_base_floorf( pow( pow( 10, i ), 1.0/2.0 ) );
```

Comment:
```suggestion
		len = floor( pow( pow( 10, i ), 1.0/2.0 ) );
```

Use the built-in `floor` here. No need to bring in an extra dep for this C benchmark.

Reply:
This is especially true given that we are using C99 `pow`.